### PR TITLE
fix: extract pagination logic

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link

[Refactor pagination calculations to common file](https://www.notion.so/uwblueprintexecs/47bf4727a2444bccad4a91952898b8f3?v=9658436bf9be409e90581631a7155f16&p=012412a68950460293fe31f8837943e1&pm=s)

Follow-up to #285

- Add hook exporting common logic for paginated data
- Use common-logic hook in table and pseudo-table components

## Implementation description
* 

## Steps to test
1. View any page with the table as well as the teacher-assessment-overview page, and verifiy they look the same.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/common/table/Table.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ import {
 } from "@chakra-ui/react";
 
 import Pagination from "./Pagination";
+import usePaginatedData from "./usePaginatedData";
```
Comment (joyce-shi) [inline frontend/src/components/common/table/Table.tsx:14]: can we remove `useState` from line 1 import!

File: frontend/src/components/common/table/usePaginatedData.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { useMemo, useState } from "react";
+
+const usePaginatedData = <T>(data: T[], pageSize = 8) => {
```
Comment (joyce-shi) [inline frontend/src/components/common/table/usePaginatedData.ts]: return type for linter!

**Review by joyce-shi — APPROVED** [review]
just some linter comments, otherwise very nice

File: frontend/src/components/common/table/Table.tsx (hunk 1 above)
Comment (jfdoming) [reply frontend/src/components/common/table/Table.tsx:14]: oops yes

**Pull request merged**